python/dgl/contrib/sampling/randomwalk.py

```diff
@@ -165,4 +169,52 @@ def bipartite_single_sided_random_walk_with_restart(
             int(max_visit_counts), int(max_frequent_visited_nodes))
     return _split_traces(traces)
 
-_init_api('dgl.randomwalk', __name__)
+
+def metapath_cycle_random_walk(hg, etypes, seeds, num_traces, max_cycles):
```

Comment:
Could we let user directly give a full metapath sequence so there is no "cycle" behavior in this API? User could directly give `[A,B,A,B,A,B]` and we would follow the metapath but could stop earlier. Also, change `etypes` argument to `metapath`.

Reply:
Good point.